src/commands/selectSignImage.ts

```diff
@@ -0,0 +1,102 @@
+import * as vscode from 'vscode';
+import { logInToDockerCli } from './registries/logInToDockerCli';
+import { checkCLI } from '../utils/checkCLI';
+import { execAsync } from '../utils/execAsync';
+
+const cliName = 'notation';
+const goToNotationButton = 'Download Notation';
+const NotationURL = 'https://notaryproject.dev/docs/installation/cli/';
+const errorMessage = `Notation not found: `;
+
+//transforms the key list string into an array of objects
+async function sortOutput(Key_String_Values: string ){
+    var dataArray = new Array();
+    var keyChoices = Key_String_Values.split("\n");
+    for (let i = 1; i < keyChoices.length; i++) { 
+      const values = keyChoices[i].trim().split(/ {2,} /); 
+      var itemObject; 
+        for (let j = 0; j < 1; j++) { 
+            const value = values[j];
+            if (value.indexOf('*')> -1){
+                const keyWithoutAsterisk = value.replace('* ', '');
+                itemObject = {label: keyWithoutAsterisk, description: "Current default"};
+            } else {
+                itemObject = {label: value};
+            }
+        }
+      dataArray.push(itemObject);
+    }
+    return dataArray;
+}
+
+//adds precreated azure key to notation
+async function addNewKey() { 
+    var key_Name = await vscode.window.showInputBox({prompt: "Input the key name"});
+    var akv_Name = await vscode.window.showInputBox({prompt: "Input the azure key vault name"});
+    if ((key_Name && akv_Name) == undefined || (key_Name && akv_Name) == '') {
+        return; // Exit the function
+    }else {
+        const progressOptions: vscode.ProgressOptions = {
+            location: vscode.ProgressLocation.Notification,
+            title: vscode.l10n.t('Adding key...'),
+        };
+        await vscode.window.withProgress(progressOptions, async () => {
+            var key_ID = await execAsync(`az keyvault certificate show -n ${key_Name} --vault-name ${akv_Name} --query "kid" -o tsv`);
```

Comment:
@Anna-Cardon I think we might want to rethink querying the Key ID from just the cert name and the AKV name. This way assumes you are using the latest version of the cert for signing. If you look at your sample AKV, you'll see that the Key ID is per version of the certificate and the `az keyvault certificate show...` command above grabs the latest version. 

I'll let others chime in to if we want to make this assumption of latest cert or we want to make it configurable. If we did want to make it configurable, I think requesting the Key ID from the user might be a better option. Of course, it requires user to have more `az`/portal interaction. 

cc: @luisdlp & @estebanreyl for their thoughts on this

Reply:
I see your point. So you are stating that users might want to use a previous version. My suggestion would be to keep this behavior by default. I don't see a lot of instances when a user would want to / should be using the older version of their certificate, is there a common scenario for this? Since I am unsure of the user scenarios in this case my suggestion is to make this configurable. Extensions can contribute additional settings and can thus make things configurable. We could add something like: "userShouldSpecifyVersion" as a config attribute that is set to false by default? Alternatively, we can ask for a version and leave note that empty defaults to "latest"?